File: leetcode/118-pascals-triangle/main.go
package main

import (
	"fmt"
)

func generate(numRows int) [][]int {
	if numRows == 0 {
		return nil
	} else if numRows == 1 {
		return [][]int{{1}}
	} else if numRows == 2 {
		return [][]int{{1}, {1, 1}}
	}

	triange := [][]int{{1}, {1, 1}}
	for i := 2; i < numRows; i++ {
		triange = append(triange, make([]int, i+1))
		triange[i][0], triange[i][len(triange[i])-1] = 1, 1
		for j := 1; j < len(triange[i])-1; j++ {
			triange[i][j] = triange[i-1][j-1] + triange[i-1][j]
		}
	}
	return triange
}

func main() {
	input := 8
	fmt.Println(generate(input))
}

File: leetcode/118-pascals-triangle/main.py
from typing import List


class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        result = [[1 for _ in range(i+1)] for i in range(numRows)]
        for i in range(2, numRows):
            for j in range(1, i):
                result[i][j] = result[i-1][j-1] + result[i-1][j]
        return result


solution = Solution()
print(solution.generate(5))
